src/responder/component.ts

```diff
@@ -232,6 +242,11 @@ export class MultiResponderComponent extends Component<ResponderAnchorState, Blo
             case ResponderActionKind.EndResponse:
                 await this.responder.endResponse(action.appointmentId);
                 break;
+            case ResponderActionKind.ResponderLowBalance:
+                if((await this.responder.signer.provider!.getBalance(this.responder.address)).gt("500000000000000000")){
```

Comment:
I think you can package this up into a function on the responder called `checkBalance`, which internally checks balance and logs if necessary. From here you would then just call `await responder.checkBalance()`.